In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np
%matplotlib inline

# Lines below are just to ignore warnings
import warnings
warnings.filterwarnings('ignore')

# Loading file

In [4]:
with open("nba_2013.csv", 'r') as csvfile: nba = pandas.read_csv(csvfile)

In [5]:
nba.head()

player pos  age bref_team_id   g  gs    mp   fg   fga    fg.  ...  \
0     Quincy Acy  SF   23          TOT  63   0   847   66   141  0.468  ...   
1   Steven Adams   C   20          OKC  81  20  1197   93   185  0.503  ...   
2    Jeff Adrien  PF   27          TOT  53  12   961  143   275  0.520  ...   
3  Arron Afflalo  SG   28          ORL  73  73  2552  464  1011  0.459  ...   
4  Alexis Ajinca   C   25          NOP  56  30   951  136   249  0.546  ...   

   drb  trb  ast  stl  blk  tov   pf   pts     season  season_end  
0  144  216   28   23   26   30  122   171  2013-2014        2013  
1  190  332   43   40   57   71  203   265  2013-2014        2013  
2  204  306   38   24   36   39  108   362  2013-2014        2013  
3  230  262  248   35    3  146  136  1330  2013-2014        2013  
4  183  277   40   23   46   63  187   328  2013-2014        2013  

[5 rows x 31 columns]

In [9]:
print('NBA Data - rows:' , nba.shape[0],'columns:', nba.shape[1])

NBA Data - rows: 481 columns: 31


In [6]:
nba.dtypes

player           object
pos              object
age               int64
bref_team_id     object
g                 int64
gs                int64
mp                int64
fg                int64
fga               int64
fg.             float64
x3p               int64
x3pa              int64
x3p.            float64
x2p               int64
x2pa              int64
x2p.            float64
efg.            float64
ft                int64
fta               int64
ft.             float64
orb               int64
drb               int64
trb               int64
ast               int64
stl               int64
blk               int64
tov               int64
pf                int64
pts               int64
season           object
season_end        int64
dtype: object

In [10]:
# Check % of null values in df
total = nba.isnull().sum().sort_values(ascending = False)
percent = (nba.isnull().sum()/nba.isnull().count()*100).sort_values(ascending = False)
pd.concat([total, percent], axis=1, keys=['Total', 'Percent']).transpose()

x3p.        ft.      x2p.     fg.    efg.  x2p  pos  age  \
Total    67.000000  20.000000  3.000000  2.0000  2.0000  0.0  0.0  0.0   
Percent  13.929314   4.158004  0.623701  0.4158  0.4158  0.0  0.0  0.0   

         bref_team_id    g  ...  orb  drb  trb  ast  stl  blk  tov   pf  pts  \
Total             0.0  0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   
Percent           0.0  0.0  ...  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0   

         player  
Total       0.0  
Percent     0.0  

[2 rows x 31 columns]

# Observations : 
    - column with highest missing values has a 13.9% values missing ie x3p column

# Missing Values

In [11]:
missingInfo = list(nba.columns[nba.isnull().any()])
missingInfo

['fg.', 'x3p.', 'x2p.', 'efg.', 'ft.']

In [12]:
for col in missingInfo:
    missingNumber = nba[nba[col].isnull() == True].shape[0]
    print('Missing Number for Col {}: {}'.format(col , missingNumber))
else :
    print(f'Data is clean , data has no missing values in columns')

Missing Number for Col fg.: 2
Missing Number for Col x3p.: 67
Missing Number for Col x2p.: 3
Missing Number for Col efg.: 2
Missing Number for Col ft.: 20
Data is clean , data has no missing values in columns
